### PR TITLE
added-reset-transaction-feature

### PR DESCRIPTION
**Pull request by eramahatra**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | yes/no
| Tests pass?   | yes
| Fixed tickets | #...
| License       | MIT

### CODE REVIEW

File: src/DirectGateway.php [hunk 1]
```diff
@@ -68,6 +68,16 @@ public function refund(array $parameters = array())
         return $this->createRequest('\Omnipay\Payline\Message\Direct\RefundRequest', $parameters);
     }
 
+    /**
+     * @param array $parameters
+     *
+     * @return \Omnipay\Payline\Message\Direct\RefundRequest
```
Comment (neeckeloo) [inline src/DirectGateway.php]: This method returns a `Omnipay\Payline\Message\Direct\ResetRequest` class.

File: src/Message/Direct/ResetRequest.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * @link      https://github.com/ck-developer/omnipay-payline
+ * @package   omnipay-payline
+ * @license   MIT
+ * @copyright Copyright (c) 2017 eric ramahatra
+ */
+
+namespace Omnipay\Payline\Message\Direct;
+
+/**
+ * CaptureRequest.
```
Comment (neeckeloo) [inline src/Message/Direct/ResetRequest.php]: `ResetRequest` instead of `CaptureRequest`.

File: src/Message/Direct/ResetRequest.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+ *
+ * @method RefundResponse send()
```
Comment (neeckeloo) [inline src/Message/Direct/ResetRequest.php]: `ResetResponse`

File: src/Message/Direct/ResetRequest.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    {
+        $data = $this->getBaseData();
+        $data['transactionID'] = $this->getTransactionReference();
+        return $data;
+    }
+
+    /**
+     * @param \stdClass $data
+     *
+     * @return RefundResponse
```
Comment (neeckeloo) [inline src/Message/Direct/ResetRequest.php]: `ResetResponse`

File: src/Message/Direct/ResetRequest.php [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+     */
+    protected function createResponse($data)
+    {
+        return $this->response = new RefundResponse($this, $data);
```
Comment (neeckeloo) [inline src/Message/Direct/ResetRequest.php]: `ResetResponse`

File: src/Message/Direct/ResetResponse.php [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+ * @link      https://github.com/ck-developer/omnipay-payline
+ * @package   omnipay-payline
+ * @license   MIT
+ * @copyright Copyright (c) 2016 - 217 Claude Khedhiri <claude@khedhiri.com>
+ */
+
+namespace Omnipay\Payline\Message\Direct;
+
+/**
+ * RefundResponse.
```
Comment (neeckeloo) [inline src/Message/Direct/ResetResponse.php]: `ResetResponse`

**Review by neeckeloo — CHANGES_REQUESTED** [review]

**Comment by ck-developer** [comment]
Hello @eramahatra, thank you for contribution. can you add the tests for this feature ?

**Pull request merged**